code/modules/awaymissions/gateway.dm

```diff
@@ -2,106 +2,191 @@ GLOBAL_DATUM(the_gateway, /obj/machinery/gateway/centerstation)
 
 /obj/machinery/gateway
 	name = "gateway"
-	desc = "A mysterious gateway built by unknown hands, it allows for faster than light travel to far-flung locations."
+	desc = "A gateway built for quick travel between linked destinations."
 	icon = 'icons/obj/machines/gateway.dmi'
 	icon_state = "off"
 	density = TRUE
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
-	var/active = 0
+	var/active = FALSE
 	var/checkparts = TRUE
-	var/list/obj/effect/landmark/randomspawns = list()
-	var/calibrated = TRUE
-	var/list/linked = list()
-	var/can_link = FALSE	//Is this the centerpiece?
+	var/list/adjacent_parts = list()
+	var/centerpiece = FALSE	//Is this the centerpiece?
+
+	/// The gateway this machine is linked to
+	var/obj/machinery/gateway/linked_gateway
+
+	/// Cooldown for says and buzz-sigh
+	COOLDOWN_DECLARE(telegraph_cooldown)
 
 /obj/machinery/gateway/Initialize(mapload)
-	randomspawns = GLOB.awaydestinations
-	update_icon()
-	if(!istype(src, /obj/machinery/gateway/centerstation) && !istype(src, /obj/machinery/gateway/centeraway))
+	. = ..()
+	if(!centerpiece)
 		switch(dir)
 			if(SOUTH,SOUTHEAST,SOUTHWEST)
 				density = FALSE
+
+/obj/machinery/gateway/Destroy()
+	if(GLOB.the_gateway == src)
+		GLOB.the_gateway = null
+	if(linked_gateway)
+		linked_gateway.linked_gateway = null
+		linked_gateway = null
 	return ..()
 
-/obj/machinery/gateway/proc/toggleoff()
-	for(var/obj/machinery/gateway/G in linked)
-		G.active = 0
-		G.update_icon()
-	active = 0
-	update_icon()
+/obj/machinery/gateway/Bumped(atom/movable/AM)
+	self_teleport(AM)
+
+/obj/machinery/gateway/MouseDrop_T(atom/movable/AM, mob/user)
+	. = ..()
+	if(AM == user)
+		self_teleport(AM) // This is so that if you're drag-clicking yourself into the gateway it'll appear as if you're entering it
+		return
 
-/obj/machinery/gateway/proc/detect()
-	if(!can_link)
+	var/turf/dest_turf = get_step(get_turf(linked_gateway), SOUTH)
+	if(!pre_check_teleport(AM, dest_turf))
+		return
+
+	if(ismob(AM))
+		var/mob/M = AM
+		user.visible_message(\
+			"<span class='warning'>[user] tries to shove [M] into [src]...</span>",\
+			"<span class='warning'>You try to shove [M] into [src]...</span>",
+			ignored_mobs = list(M))
+		to_chat(M, "<span class='userdanger'>[user] is pushing you into [src]!</span>")
+		if(do_after(user, 5 SECONDS, src))
+
+		else
+			return // failed do_after, we don't teleport
+	else
+		user.visible_message("<span class='notice'>[AM] is pushed into [src].</span>") //
```

Comment:
There is a comment here saying nothing. There is also no self_message

Reply:
oversight